src/main/java/com/redhat/devtools/intellij/tektoncd/actions/triggers/CreateTriggerBindingAction.java

```diff
@@ -13,25 +13,53 @@
 import com.google.common.base.Strings;
 import com.intellij.openapi.actionSystem.AnActionEvent;
 import com.redhat.devtools.intellij.tektoncd.actions.TektonAction;
+import com.redhat.devtools.intellij.tektoncd.telemetry.TelemetryService;
 import com.redhat.devtools.intellij.tektoncd.tkn.Tkn;
 import com.redhat.devtools.intellij.tektoncd.tree.TriggerBindingsNode;
 import com.redhat.devtools.intellij.tektoncd.utils.VirtualFileHelper;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 import javax.swing.tree.TreePath;
 
+import java.io.IOException;
+
+import static com.redhat.devtools.intellij.tektoncd.Constants.KIND_CONDITIONS;
 import static com.redhat.devtools.intellij.tektoncd.Constants.KIND_TRIGGERBINDINGS;
+import static com.redhat.devtools.intellij.tektoncd.telemetry.TelemetryService.PROP_RESOURCE_KIND;
+import static com.redhat.devtools.intellij.telemetry.core.service.TelemetryMessageBuilder.ActionMessageBuilder;
+import static com.redhat.devtools.intellij.telemetry.core.util.AnonymizeUtils.anonymizeResource;
 
 public class CreateTriggerBindingAction extends TektonAction {
+
+    private static final Logger logger = LoggerFactory.getLogger(CreateTriggerBindingAction.class);
+
     public CreateTriggerBindingAction() { super(TriggerBindingsNode.class); }
 
     @Override
     public void actionPerformed(AnActionEvent anActionEvent, TreePath path, Object selected, Tkn tkncli) {
+        ActionMessageBuilder telemetry = TelemetryService.instance()
+                .action("create trigger binding")
+                .property(PROP_RESOURCE_KIND, KIND_TRIGGERBINDINGS);
         TriggerBindingsNode item = getElement(selected);
         String namespace = item.getParent().getName();
         String content = getSnippet("Tekton: TriggerBinding");
 
-        if (!Strings.isNullOrEmpty(content)) {
-            VirtualFileHelper.createAndOpenVirtualFile(anActionEvent.getProject(), namespace, namespace + "-newtriggerbinding.yaml", content, KIND_TRIGGERBINDINGS, item);
+        if (Strings.isNullOrEmpty(content)) {
+            telemetry
+                    .error("snippet content empty.")
+                    .send();
+        } else {
+            String name = namespace + "-newtriggerbinding.yaml";
+            try {
+                VirtualFileHelper.createAndOpenVirtualFile(anActionEvent.getProject(), namespace, name, content, KIND_TRIGGERBINDINGS, item);
+                telemetry.send();
```

Comment:
this should be `telemetry.success().send()`

Reply:
`success()` is not required any longer. The builder would add it automatically if there's no result or error when sending:
https://github.com/redhat-developer/intellij-redhat-telemetry/blob/main/src/main/java/com/redhat/devtools/intellij/telemetry/core/service/TelemetryMessageBuilder.java#L179
```
        private void ensureResultOrError() {
            if (!hasProperty(PROP_ERROR)
                    && !hasProperty(PROP_RESULT)) {
                success();
            }
        }
```